indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,309 @@
-        {({fields}) => {
-          if (!fields.value) {
-            return null;
-          }
-          const options = this.generateAttributeOptions(attributes, fields.value);
-          return (
-            <div>
-              <Dropdown
-                className="icon room-edit-modal-add-btn"
-                button
-                text={content.placeholder}
-                floating
-                labeled
-                icon="add"
-                options={options}
-                search
-                disabled={options.length === 0}
-                selectOnBlur={false}
-                value={null}
-                onChange={(__, {value: name}) => {
-                  fields.push({value: null, name});
-                }}
-              />
-              {fields.map((attribute, index) => (
-                <div key={attribute}>
-                  <FinalInput
-                    name={`${attribute}.value`}
-                    label={titles[fields.value[index].name]}
-                    required
-                    icon={{
-                      name: 'remove',
-                      color: 'red',
-                      link: true,
-                      onClick: () => fields.remove(index),
-                    }}
-                  />
-                </div>
-              ))}
-              {fields.length === 0 && (
-                <div>
-                  <Translate>No custom attributes found</Translate>
-                </div>
-              )}
-            </div>
-          );
-        }}
-      </FieldArray>
-    );
   };
 
-  renderContent = (content, key) => {
-    const {roomEquipment, room} = this.state;
-    const {equipmentTypes, roomId} = this.props;
-    const hasPhoto = this.newRoom ? false : room.hasPhoto;
-    switch (content.type) {
-      case 'header':
-        if (content.key === 'photo' && this.newRoom) {
-          // XXX using the key for this is awful..
-          return null;
-        }
-        return <Header key={key}>{content.label}</Header>;
-      case 'input': {
-        return (
-          <FinalInput
-            key={key}
-            name={content.name}
-            label={content.label}
-            required={content.required}
-            parse={content.parse}
-            {...content.inputArgs}
-          />
-        );
+  const closeModal = useCallback(async () => {
+    setLoading(true);
+    if ((roomId || newRoomId) !== null && wasEverUpdated) {
+      await Promise.all([
+        dispatch(roomsActions.fetchRoom(roomId)),
+        dispatch(roomsActions.fetchEquipmentTypes()),
+        dispatch(userActions.fetchRoomPermissions(roomId)),
+        dispatch(roomsActions.fetchDetails(roomId, true)),
+      ]);
+    }
+    onClose(wasEverUpdated || afterCreation);
+  }, [dispatch, onClose, roomId, newRoomId, wasEverUpdated, afterCreation]);
+
+  useEffect(() => {
+    (async () => setGlobalAttributes(await fetch(fetchAttributesURL())))();
```

Comment:
This only runs once since `fetch` is defined with an empty dependency list and thus never changes, right?

If yes you could probably use `useIndicoAxios`. And by no longer having a dependency on your `fetch` util here it means that you may not need to wrap that util in `useCallback` anymore.

Reply:
Exactly, we can actually remove the `fetch` dependency here.

Reply:
Actually, `useIndicoAxios` doesn't seem to provide the error handling we have in fetch, does it?
Unless we have a utils method for it, I would like to avoid having more than one line here.

Reply:
It triggers the error dialog automatically in case of an error, and returns `{data, error, loading}` (and some other) things